temporian/implementation/numpy/operators/tick_calendar.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Implementation for the TickCalendar operator."""
+
+from datetime import datetime, timedelta
+from typing import Dict, List
+
+import numpy as np
+
+from temporian.implementation.numpy.data.event_set import IndexData, EventSet
+from temporian.core.operators.tick_calendar import TickCalendar
+from temporian.implementation.numpy import implementation_lib
+from temporian.implementation.numpy.operators.base import OperatorImplementation
+from temporian.core.data import duration
+from temporian.implementation.numpy_cc.operators import operators_cc
+
+
+class TickCalendarNumpyImplementation(OperatorImplementation):
+    def __init__(self, operator: TickCalendar) -> None:
+        assert isinstance(operator, TickCalendar)
+        super().__init__(operator)
+
+    def __call__(self, input: EventSet) -> Dict[str, EventSet]:
+        assert isinstance(self.operator, TickCalendar)
+        output_schema = self.output_schema("output")
+
+        # Create output EventSet
+        output_evset = EventSet(data={}, schema=output_schema)
+
+        # Fill output EventSet's data
+        for index_key, index_data in input.data.items():
+            if len(index_data.timestamps) < 2:
+                dst_timestamps = np.array([], dtype=np.float64)
```

Comment:
Add a note in the documentation telling that, if there is a single event that fall exactly on a calendar event specified by the user, the tick_calendar op will *not* generate a timestamp.

I think it would be simpler if a single event would be begin = end = single event timestamp.

Note: That an event set can contain multiple events with the same timestamp value.

Reply:
You're right, since this means closed interval, it makes sense to allow a single value.